### ISSUE TITLE
Reduced readability from clippy suggestion.

### ISSUE THREAD

**Issue by continue-100** [rust-lang/rust-clippy#17133]
### Summary

`!(a > b)` is to me, more readable in certain contexts, yet clippy unnecessarily asks to simplify it to `a <= b`.

### Lint Name

nonminimal_bool

### Reproducer

```
warning: this boolean expression can be simplified
 --> src/main.rs:9:16
  |
9 |             if !(a > b) && !descent_regime {}
  |                ^^^^^^^^ help: try: `(a <= b)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.96.0/index.html#nonminimal_bool
  = note: `#[warn(clippy::nonminimal_bool)]` on by default

```

### Version

```text

```

**Comment by samueltardieu** [comment]
Starting from Rust 1.97 (currently in beta), `nonminimal_bool` will be a `pedantic` lint which doesn't trigger in the default configuration.

@rustbot label -C-bug -I-false-positive
